src/Generator/TypeGenerator/BaseComplexType.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Laminas\Code\Generator\TypeGenerator;
+
+abstract class BaseComplexType implements Type
```

Comment:
`/** @internal */` for anything we didn't build explicitly for external systems

Reply:
s/`BaseComplexType`/`CompositeType`? 

We already know it's a "base" type from the `abstract` keyword